setup.py

```diff
@@ -1,6 +1,16 @@
-from distutils.core import setup
+#! /usr/bin/env python
+"""Distribution setup module."""
 
-setup(name='pysize',
-      version='1.0',
-      py_modules=['pysize'],
-      )
\ No newline at end of file
+__requires__ = ('setuptools >=36.3.0', )

```

Comment:
@bosswissam do you think you want to require this version of setuptools as a pre-requisite?

Reply:
what's your recommendation? Sorry not super familiar with pypi setup.

Reply:
Well, the question is what do you want for your users.

You're going to have a wheel (binary dist) and a tarball (source dist).
When building wheel, it's okay to upgrade setuptools in CI to use latest features.

But when it comes to users they may have older setuptools on their machines.
It doesn't affect wheel, because its installation is just unpacking a zip archive.
With sdist it's a bit more complicated: `pip install` actually executes `setup.py install` under the hood. So all those options/meta stored in `setup.cfg` will simply not be read. Of course, I've got a shim for that if needed.

But most of the users will install from wheel anyway.

The decision is up to you.

Reply:
@bosswissam so?

Reply:
@bosswissam ^